tools/run-container

```diff
@@ -463,9 +463,9 @@ main() {
         return
     }
 
-    inside_as_cd "$name" root "$cdir" \
-        python3 ./tools/read-dependencies "--distro=${OS_NAME}" \
-            --test-distro || {
+    local rdcmd=(python3 tools/read-dependencies "--distro=${OS_NAME}" --install)
+    [[ ${OS_NAME} =~ (ubuntu|debian) ]] && rdcmd+=('--test-distro')
```

Comment:
The immediate reason why I'm passing `--test-distro` only on ubuntu|debian is that `python3-tox` is not available on CentOS.

Anyway `--test-distro` only does two thing on top on what is done without it:

 1. Adds the packages defined in `CI_SYSTEM_BASE_PKGS` to the dependencies
 2. Actually installs the packages on the system

and in my understanding is meant to enable testing of the CI automation, something we are doing only on Ubuntu anyway. The deps in `test-requirements.txt` are still installed as `brpm` and `bddeb` explicitly make `read-dependencies` read it, so the unit testing can still be done even without `--test-distro` provided that all the deps are available (not the case on CentOS, but different story).

Reply:
I guess if you're trying to make this work despite lack of available deps on centos (7?) then the change/knowledge to ignore should go into tools/read-dependencies.  Because if someone goes to fix this, they'll now have to remember to add the '--test-distro' flag back here, *and* fix the tools/read-dependencies to do have the right packages.

but make this is fine. thanks for explaining.

we really do need to have the ability to run test on linux other than ubuntu "really easily". Hopefully @OddBloke's pytest will land and make that possible?

Reply:
@smoser thanks for your feedback here.

Deps are missing in CentOS 8 too (at least: `python3-tox`, `python3-contextlib2`, `python3-httpretty`, and `python3-unittest2` is only available in the PowerTools repo).

I'm not sure I agree the "install the test deps on ubuntu/debian only" logic should go in `read-dependencies`. If I leave `run-container` as it is it will always call `read-dependencies --test-distro`, which if the distro is (say) CentOS 8 can do two things:

1. Try to setup the test environment and fail, because dependencies are missing. This is consistent with everything, but not good for us, as we need to run the thing.
2. Skip setting up the test environment if distro != ubuntu|debian (or similar conditions). This is lying to the user, isn't it? A test environment has been asked for, but it hasn't been set up.

I think that not asking for a test environment (that is, not passing `--test-distro`) when we know it can't be set up is the correct thing to do.

Reply:
@paride Now that pytest change has landed #211 does that mean we can run tests on Centos8?  or are there still deps missing?

> I'm not sure I agree the "install the test deps on ubuntu/debian only" logic should go in read-dependencies. If I leave run-container as it is it will always call read-dependencies --test-distro, which if the distro is (say) CentOS 8 can do two things:

I'd be OK not passing --test-distro to Centos8 as long as we logged a warning that we skip it due to the fact that it's currently broken.  We should open a bug for centos8 test-enviroment  as an upstream bug and mention that in the warning.

Reply:
There are still deps missing: contextlib2, httpretty, unittest2. Of these unittest2 is available in the PowerTools repository, the other two are not (according to pkgs.org). I'm +1 with opening a bug and mentioning it in a warning message.

Reply:
@raharper I had a look at `https://www.rdoproject.org/what/repos/` but it seems that it only has python2 versions of the libraries we need, and no httpretty. I enabled both the openstack-train and openstack-train-testing repos, which should have the most up to date packages.

```
[root@centos8 ~]# yum repolist
Last metadata expiration check: 0:04:41 ago on Thu Mar 12 18:36:28 2020.
repo id                                                              repo name                                                                                            status
AppStream                                                            CentOS-8 - AppStream                                                                                 5103
BaseOS                                                               CentOS-8 - Base                                                                                      2107
*epel                                                                Extra Packages for Enterprise Linux 8 - x86_64                                                       4979
*epel-modular                                                        Extra Packages for Enterprise Linux Modular 8 - x86_64                                                  0
extras                                                               CentOS-8 - Extras                                                                                       3
openstack-train                                                      OpenStack Train Repository                                                                           2212
openstack-train-testing                                              OpenStack Train Testing                                                                              2236
rdo-qemu-ev                                                          RDO CentOS-7 - QEMU EV                                                                                 33

[root@centos8 ~]# yum --disablerepo="*" --enablerepo="openstack-train" list available | grep -E '(contextlib2|unittest2|httpretty)'
python2-contextlib2.noarch                            0.5.5-8.el7                   openstack-train
python2-unittest2.noarch                              1.1.0-15.el7                  openstack-train

[root@centos8 ~]# yum --disablerepo="*" --enablerepo="openstack-train-testing" list available | grep -E '(contextlib2|unittest2|httpretty)'
python2-contextlib2.noarch                            0.5.5-8.el7                   openstack-train-testing
python2-unittest2.noarch                              1.1.0-15.el7                  openstack-train-testing
```